### DISCUSSION TITLE
How Langchain-go handles when the prompt exceed context length

### DISCUSSION THREAD

**Discussion by vsumit89** [tmc/langchaingo#317] [Q&A]
Hi there,

I've been using Langchain-go and have a quick question about how it handles prompts that exceed the defined context length. Can someone please provide a brief overview or point me to the relevant code/documentation on this topic?

Thanks in advance!

**Comment by jcantonio** [comment]
Hi @vsumit89 ,
I believe that Langchain uses ConversationSummaryMemory which uses a chain to summarise the conversation limit the size of the context (SystemMessage).
I could not see it in Langchaingo though.
